### PR TITLE
mh: write test case log messages into artifacts directory

### PR DESCRIPTION
**Pull request by pbrezina**
If no log path is set via `--mh-log-path`, the produces messages are
automatically written into artifacts directory if any artifacts should be
generated (depends on `--mh-collect-artifacts`).

### CODE REVIEW

File: pytest_mh/_private/fixtures.py [hunk 1]
```diff
@@ -249,26 +249,51 @@ def _teardown(self) -> None:
...
+        path = self._artifacts_dir()
+        if path is None:
+            return
+
+        host.collect_artifacts(path)
+
+    def _flush_logs(self) -> None:
+        """
+        Write log messages produces by current test case to a file or clear
+        them if not artifacts should be generated.
```
Comment (justin-stephenson) [inline pytest_mh/_private/fixtures.py:290]: Minor grammar fixes
~~~
Write log messages produced by current test case to a file, or clear them if no artifacts should be generated.
~~~

**Comment by justin-stephenson** [comment]
Are the logs written to these new files inside `artifacts` intended to be the same as what is written to `--mh-log-path` log file? The logs generated when using `--mh-log-path` contain several more log messages than the ones written to setup.log and teardown.log.
~~~
(.venv) [justin@fedora system]$ cat artifacts/setup.log
INFO     2023-11-13 09:45:02,710 Creating SSSD systemd override
INFO     2023-11-13 09:45:02,710 Opening SSH connection to client.test
INFO     2023-11-13 09:57:56,231 Creating SSSD systemd override
INFO     2023-11-13 09:57:56,231 Opening SSH connection to client.test
INFO     2023-11-13 09:59:22,978 Creating SSSD systemd override
INFO     2023-11-13 09:59:22,978 Opening SSH connection to client.test
~~~

Also I see only a single `artifacts/setup.log`  and `artifacts/teardown.log` when multiple tests are executed, how would someone differentiate the logs when multiple tests are executed? Or do these logs only pertain to setup and teardown of the entire test sequence, not individual tests?

**Comment by pbrezina** [comment]
> Are the logs written to these new files inside `artifacts` intended to be the same as what is written to `--mh-log-path` log file? The logs generated when using `--mh-log-path` contain several more log messages than the ones written to setup.log and teardown.log.
> 
> ```
> (.venv) [justin@fedora system]$ cat artifacts/setup.log
> INFO     2023-11-13 09:45:02,710 Creating SSSD systemd override
> INFO     2023-11-13 09:45:02,710 Opening SSH connection to client.test
> INFO     2023-11-13 09:57:56,231 Creating SSSD systemd override
> INFO     2023-11-13 09:57:56,231 Opening SSH connection to client.test
> INFO     2023-11-13 09:59:22,978 Creating SSSD systemd override
> INFO     2023-11-13 09:59:22,978 Opening SSH connection to client.test
> ```
> 
> Also I see only a single `artifacts/setup.log` and `artifacts/teardown.log` when multiple tests are executed, how would someone differentiate the logs when multiple tests are executed? Or do these logs only pertain to setup and teardown of the entire test sequence, not individual tests?

It is split into individual tests, use `--mh-collect-artifacts=always` to collect it also for green tests. Then you will have:

```
artifacts/setup.log
artifacts/teardown.log
artifacts/$testcase/test.log
```

File: pytest_mh/_private/fixtures.py (hunk 1 above)
Comment (pbrezina) [reply pytest_mh/_private/fixtures.py]: Fixed, thank you.

**Pull request merged**